File: src/test/scala/repository/eventsourcing/example/domain/HouseBuilt.scala
package repository.eventsourcing.example.domain

import repository.eventsourcing.InitialEvent

case class HouseBuilt(address: String, priceInCents: Long, owner: String) extends InitialEvent[House] {
  override def initializedObject(): House =
    House.build(address, priceInCents, owner)
}

File: ddd-core/src/test/java/repository/eventsourcing/example/domain/HouseBuilt.java
package repository.eventsourcing.example.domain;

import repository.eventsourcing.InitialEvent;

public class HouseBuilt extends InitialEvent<House> {

    public final String address;
    public final long priceInCents;
    public final String owner;

    public HouseBuilt(String address, long priceInCents, String owner) { 
        this.address = address;
        this.priceInCents = priceInCents;
        this.owner = owner;
    }

    public House initializedObject() { return new House(address, priceInCents, owner); }
}
